src/testing/mock-web-locks.ts

```diff
@@ -0,0 +1,58 @@
+import assert from "node:assert";
+import Lock from "../vendor/browser-tabs-lock";
+
+/**
+ * Neither Node or JSDOM provide an implementation of the Web Locks API, so we
+ * implement a mock in terms of a locking implementation we do have on hand.
+ */
+export class MockWebLocks implements LockManager {
```

Comment:
In some ways this feels a bit silly, but at least it allows us to exercise both code paths without needing to bring in another dependency or implement the entirety of a lock implementation ourselves.